src/internal/connector/exchange/service_iterators.go

```diff
@@ -185,6 +190,13 @@ func filterContainersAndFillCollections(
 		graph.NewMetadataEntry(graph.PreviousPathFileName, currPaths),
 	}
 
+	logger.Ctx(ctx).Infow(
+		"persisted metadata",
+		"num_paths_entries",
+		len(currPaths),
+		"num_deltas_entries",
+		len(deltaURLs))
```

Comment:
Nit: can I push for keeping key-value pairs on the same line?  Kinda reads like a map that way, which is easy to grok.
```suggestion
		"num_paths_entries", len(currPaths),
		"num_deltas_entries", len(deltaURLs))
```